nixos/modules/system/boot/initrd-network.nix

```diff
@@ -39,6 +39,9 @@ let
       fi
     '';
 
+  udhcpcHostnameArg = if cfg.udhcpc.hostName != ""
```

Comment:
I would be more in favor of a `list`-style `extraArgs` or `settings` and having the hostname option write into it.